.github/workflows/build_publish_lambda_layer.yml

```diff
@@ -95,18 +143,10 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
-      - if: ${{ inputs.publish-dest == 'staging' }}
-        name: use staging credentials
-        uses: aws-actions/configure-aws-credentials@v4
-        with:
-          role-to-assume: ${{ secrets.LAMBDA_PUBLISHER_ARN_STAGING }}
-          aws-region: ${{ matrix.aws_region }}
-
-      - if: ${{ inputs.publish-dest == 'production' }}
-        name: use production credentials
+      - name: configure AWS credential
         uses: aws-actions/configure-aws-credentials@v4
         with:
-          role-to-assume: ${{ secrets.LAMBDA_PUBLISHER_ARN_PROD }}
+          role-to-assume: ${{ inputs.publish-dest == 'production' && secrets.LAMBDA_PUBLISHER_ARN_PROD || inputs.publish-dest == 'staging' && secrets.LAMBDA_PUBLISHER_ARN_STAGING }}
```

Comment:
Clever... but now we've lost that info regarding whether we're publishing to staging or production. Can the stage/prod info be somewhere like the step name or output to workflow logs?

Reply:
Updated to step name.